### ISSUE TITLE
There's no way to test multiple nameservers

### ISSUE THREAD

**Issue by Paladin** [spotify/rspec-dns#22]
for example:

    it "should properly resolve internally" do
      expect(domain).to have_dns.config(nameserver: 'internal').
        with_type('A').
        and_address(address)
    end
    it "should properly resolve externally" do
      expect(domain).to have_dns.config(nameserver: 'external').
        with_type('A').
        and_address(address)
    end

given that 'internal' and 'external' are different nameservers (and are not properly sync'ed, so they have different addresses for 'domain') my 'expected behavior' is for one test to pass and the other to fail.

Instead what seems to be happening is they both either pass or fail, depending upon the state of 'internal': if 'internal' has the expected address, while 'external' has the wrong one, they both pass. Reverse the situation and they both fail.

**Comment by otahi** [comment]
@Paladin, thank you for reporting!

I will check it.

**Comment by otahi** [comment]
I have found something. I will investigate more.

```ruby
require 'rspec-dns'

%w(8.8.8.8 8.8.4.4).each do |s|
  describe s do
    it do
      expect('www.google.com').to have_dns.with_type('A')
        .config(nameserver: s)
    end
  end
end
```

Then got
```
[otahi@otahiair rspec-dns]$ bundle exec rspec spec/test.rb

8.8.8.8
  should have the correct dns entries with {:type=>"A"}

8.8.4.4
  should have the correct dns entries with {:type=>"A"}

Finished in 0.12608 seconds (files took 0.20845 seconds to load)
2 examples, 0 failures
[otahi@otahiair rspec-dns]$
```

But tcpdump shows no 8.8.4.4

```
[otahi@otahiair ~]$ sudo tcpdump -nnn port 53
tcpdump: data link type PKTAP
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on pktap, link-type PKTAP (Packet Tap), capture size 65535 bytes
23:26:41.725576 IP 192.168.11.3.53467 > 8.8.8.8.53: 44501+% [1au] ANY? www.google.com. (43)
23:26:41.740006 IP 8.8.8.8.53 > 192.168.11.3.53467: 44501 2/0/1 A 216.58.220.228, AAAA 2404:6800:4004:814::2004 (87)
^C
2 packets captured
8 packets received by filter
0 packets dropped by kernel
[otahi@otahiair ~]$
```

**Comment by otahi** [comment]
I have found the solution.

Resolver has `do_caching` attribute on [dnsruby/resolver.rb#L177](https://github.com/alexdalitz/dnsruby/blob/master/lib/dnsruby/resolver.rb#L177).

Rspec-dns does not set `do_caching` and default value is `true`.
Rspec-dns is a tool to test dns servers, so `do_caching` should be set to `false`.

When I set `do_caching` `false`, I can see 2 queries.

```
[otahi@otahiair ~]$ sudo tcpdump -nnn port 53
tcpdump: data link type PKTAP
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on pktap, link-type PKTAP (Packet Tap), capture size 65535 bytes
23:45:45.250002 IP 192.168.11.3.51239 > 8.8.8.8.53: 4785+% [1au] ANY? www.google.com. (43)
23:45:45.538507 IP 8.8.8.8.53 > 192.168.11.3.51239: 4785 2/0/1 A 216.58.220.228, AAAA 2404:6800:4004:814::2004 (87)
23:45:45.542486 IP 192.168.11.3.63653 > 8.8.4.4.53: 40847+% [1au] ANY? www.google.com. (43)
23:45:45.604355 IP 8.8.4.4.53 > 192.168.11.3.63653: 40847 2/0/1 A 216.58.220.228, AAAA 2404:6800:4004:814::2004 (87)
^C
4 packets captured
20 packets received by filter
0 packets dropped by kernel
[otahi@otahiair ~]$
```

@Paladin, Can you try [this change](https://github.com/otahi/rspec-dns/blob/change_resolver_not_to_do_chaching/lib/rspec-dns/have_dns.rb#L155) on your environment? If it looks good to you, I will create a pull request.

**Comment by Paladin** [comment]
If you mean:

it "should properly resolve internally" do
  expect(domain).to have_dns.config(nameserver: 'internal').config(do_caching: false).
    with_type('A').
    and_address(address)
end
it "should properly resolve externally" do
  expect(domain).to have_dns.config(nameserver: 'external').config(do_caching: false).
    with_type('A').
    and_address(address)
end

Then no. The results are still different depending upon which nameserver comes first. I'll play around with some other approaches to it to see if they make it work.

**Comment by Paladin** [comment]
OK, trying that change improves matters, but I still get different results depending upon which is first. Let me play a bit with the timeouts, though, I may be having timeout/connection issues as well, because I will occasionally get different results for successive runs of the same file.

**Comment by otahi** [comment]
@Paladin,  I'm sorry. I have not to tell you how try it.

I have changed `have_dns` method to set `do_caching` `false`at [lib/rspec-dns/have_dns.rb#L155](https://github.com/otahi/rspec-dns/blob/change_resolver_not_to_do_chaching/lib/rspec-dns/have_dns.rb#L155) on my branch.

If you use bundler, please change like as follows.

```
source 'https://rubygems.org'

gem 'rspec-dns'
```
:arrow_down: 
```
source 'https://rubygems.org'

gem 'rspec-dns', git: 'https://github.com/otahi/rspec-dns.git', branch: 'change_resolver_not_to_do_chaching'
```

After `bundle install`, you can do your original code without any modifications.

**Comment by Paladin** [comment]
OK, after hammering a bit on the proposed change, I think that's got it. I'm still having consistency issues, but the issues are always with the same nameserver now, regardless of the order of the queries, so I'm figuring the problems that remain are mine, not yours.

**Comment by Paladin** [comment]
Weird, I wrote that comment 3 days ago, just discovered I never hit send. Sorry about that.

**Comment by Paladin** [comment]
I didn't change the gem source, incidentally. I just edited the source file from the gem I installed. Yes, it's tacky, but I figured it was a temporary change, and only for this one gemset, so it wouldn't cause any more trouble than I could handle.

**Comment by otahi** [comment]
OK, I will create #23, please check it.

**Comment by otahi** [comment]
@Paladin, Fix was released at [0.1.6](https://rubygems.org/gems/rspec-dns/versions/0.1.6).
Please check it and close this issue if no problem.

**Comment by Paladin** [comment]
Looks good. Ran against three nameservers, switched their order so that each one had a chance to be first up. Only errors in the runs came from the flaky server, which is my problem (and why I was using this) not yours.
